cilium/CFP-39876-clustermesh-filtered-export.md

```diff
@@ -0,0 +1,110 @@
+# CFP-39876: Scoped Export Mode for ClusterMesh
+
+**SIG:** SIG-NAME
+
+**Sharing:** Public
+
+**Begin Design Discussion:** 2025-06-02
+
+**End Design Discussion:**
+
+**Cilium Release:** (TBD)
+
+**Authors:** Krunal Jain <krunaljain@microsoft.com>, Vamsi Kalapala <Krishna.Vamsi@microsoft.com>
+
+## Summary
+
+This CFP introduces selective distribution of endpoints and identities within Cilium Clustermesh by introducing a new “scoped-export” mode for the Clustermesh API server, restricting cross-cluster propagation only to resources that are fronted by global services. The modification targets improved scalability while acknowledging that direct inter-cluster endpoint access and network policy enforcement will be discontinued for resources not associated with global services. The addition of new Clustermesh APIserver export mode would ensure the changes are backwards compatible.
```

Comment:
> will be discontinued

I'd suggest clarifying what *discontinued* means in practice.

Reply:
Based on the feedback to allowlist pods for export, we can support network policies as is. I will remove this part.

Reply:
Going with the namespace allowlisting approach. Updated the goals for Network Policy support